### ISSUE TITLE
Customer notes from appointment

### ISSUE THREAD

**Issue by Chotimir** [Microsoft/bookings-samples#9]
Hello,

How can I access field 'customerNotes' from appointment? I tried using graph API (/bookingBusinesses/{id}/appointments) but it always returns null for customerNotes.

Thanks
Jakub

**Comment by kolomolo** [comment]
Hi,

@fpintos Could you help with this please?

Thanks
Alex

**Comment by fpintos** [comment]
Indeed...that property is missing as you noticed. At some point there was an issue preventing us from reading it which was recently solved; the Bookings API needs to be updated account for that.

In the meantime, the value can be found in the Graph using an extended property of the underlying event. For example:
```
https://graph.microsoft.com/beta/users/{businessId}/events/{appointmentId}/?$select=id,singleValueExtendedProperties&$expand=singleValueExtendedProperties($filter=id eq 'String {ccb88e73-951a-4b10-bb49-e6628b8d8494} Name BookingItemInternalNotes')
```

I'll keep this open until the fix is deployed.

**Comment by kolomolo** [comment]
Thanks


On Sat, Jul 7, 2018, at 12:02 AM, Fabio Pintos wrote:
> Indeed...that property is missing as you noticed. At some point there
> was an issue preventing us from reading it which was recently solved;
> the Bookings API needs to be updated account for that.> In the meantime, the value can be found in the Graph using an extended
> property of the underlying event. For example:


> https://graph.microsoft.com/beta/users/{businessId}/events/{appointmentId}/?$select=id,singleValueExtendedProperties&$expand=singleValueExtendedProperties($filter=id
> eq 'String {ccb88e73-951a-4b10-bb49-e6628b8d8494} Name
> BookingItemInternalNotes')
>> I'll keep this open until the fix is deployed.


> — You are receiving this because you commented. Reply to this email
> directly, view it on GitHub[1], or mute the thread[2].
Links:

  1. https://github.com/Microsoft/bookings-samples/issues/9#issuecomment-403157848

**Comment by fpintos** [comment]
Hi @Chotimir , the fix to populate the customerNotes has been deployed. Note that, [as stated in the docs](https://developer.microsoft.com/en-us/graph/docs/api-reference/beta/resources/bookingappointment), customerNotes, serviceNotes and reminders are properties available only when reading an individual appointment, not when querying the entity set.

That being said, as I was verifying the fix I noticed a discrepancy in the appointments created by the self-service page.

When appointments are created from OWA, the mobile apps and the API, both customerNotes and serviceNotes are set as expected. However, when the appointment is created from self service page the 'customer notes' ends up appended to the 'serviceNotes' and the original service notes are not included.

I've pointed this out to the product team for them to take a look.
